### PR TITLE
Support for Basic Authentication

### PR DESCRIPTION
**Pull request by lxhoan**

### CODE REVIEW

**Comment by majst01** [comment]
LGTM

**Comment by majst01** [comment]
Hi @lxhoan would you mind adding a CHANGELOG entry ?

**Comment by lxhoan** [comment]
changelog added

**Comment by lxhoan** [comment]
with reference to InfluxDb document

https://docs.influxdata.com/influxdb/v1.6/administration/authentication_and_authorization/
Hi @fmachado, it's still backward compatible (actually in the client now Authentication scheme is transparent to user). I just propose to to switch to Basic Auth because in InfluxDB document

https://docs.influxdata.com/influxdb/v1.6/administration/authentication_and_authorization/

basic authentication "is the preferred method for providing user credentials"

**Comment by fmachado** [comment]
@lxhoan ok but you modified the existing public API by removing parameters and this will break the existing users implementation when they update the library.

File: src/main/java/org/influxdb/impl/InfluxDBService.java [hunk 1]
```diff
@@ -39,49 +39,44 @@
    *                             Can be one of one, any, all, quorum. Defaults to all.
    */
   @POST("write")
-  public Call<ResponseBody> writePoints(@Query(U) String username,
-      @Query(P) String password, @Query(DB) String database,
+  public Call<ResponseBody> writePoints(@Query(DB) String database,
```
Comment (fmachado) [inline src/main/java/org/influxdb/impl/InfluxDBService.java]: parameters removed: (_non-backward compatible change_)

File: src/main/java/org/influxdb/impl/InfluxDBService.java [hunk 2]
```diff
@@ -39,49 +39,44 @@
...
       @Query(RP) String retentionPolicy, @Query(PRECISION) String precision,
       @Query(CONSISTENCY) String consistency, @Body RequestBody batchPoints);
 
   @GET("query")
-  public Call<QueryResult> query(@Query(U) String username, @Query(P) String password, @Query(DB) String db,
+  public Call<QueryResult> query(@Query(DB) String db,
```
Comment (fmachado) [inline src/main/java/org/influxdb/impl/InfluxDBService.java]: parameters removed: (_non-backward compatible change_)

File: src/main/java/org/influxdb/impl/InfluxDBService.java [hunk 3]
```diff
@@ -39,49 +39,44 @@
...
       @Query(EPOCH) String epoch, @Query(value = Q, encoded = true) String query);
 
   @POST("query")
-  public Call<QueryResult> query(@Query(U) String username, @Query(P) String password, @Query(DB) String db,
+  public Call<QueryResult> query(@Query(DB) String db,
```
Comment (fmachado) [inline src/main/java/org/influxdb/impl/InfluxDBService.java]: parameters removed: (_non-backward compatible change_)

File: src/main/java/org/influxdb/impl/InfluxDBService.java [hunk 4]
```diff
@@ -39,49 +39,44 @@
...
           @Query(EPOCH) String epoch, @Query(value = Q, encoded = true) String query,
           @Query(value = PARAMS, encoded = true) String params);
 
   @GET("query")
-  public Call<QueryResult> query(@Query(U) String username, @Query(P) String password, @Query(DB) String db,
+  public Call<QueryResult> query(@Query(DB) String db,
```
Comment (fmachado) [inline src/main/java/org/influxdb/impl/InfluxDBService.java]: parameters removed: (_non-backward compatible change_)

File: src/main/java/org/influxdb/impl/InfluxDBService.java [hunk 5]
```diff
@@ -39,49 +39,44 @@
...
       @Query(value = Q, encoded = true) String query);
 
   @POST("query")
-  public Call<QueryResult> postQuery(@Query(U) String username, @Query(P) String password, @Query(DB) String db,
+  public Call<QueryResult> postQuery(@Query(DB) String db,
```
Comment (fmachado) [inline src/main/java/org/influxdb/impl/InfluxDBService.java]: parameters removed: (_non-backward compatible change_)

File: src/main/java/org/influxdb/impl/InfluxDBService.java [hunk 6]
```diff
@@ -39,49 +39,44 @@
...
       @Query(value = Q, encoded = true) String query);
 
   @POST("query")
-  public Call<QueryResult> postQuery(@Query(U) String username, @Query(P) String password, @Query(DB) String db,
+  public Call<QueryResult> postQuery(@Query(DB) String db,
```
Comment (fmachado) [inline src/main/java/org/influxdb/impl/InfluxDBService.java]: parameters removed: (_non-backward compatible change_)

File: src/main/java/org/influxdb/impl/InfluxDBService.java [hunk 7]
```diff
@@ -39,49 +39,44 @@
...
           @Query(value = Q, encoded = true) String query, @Query(value = PARAMS, encoded = true) String params);
 
   @GET("query")
-  public Call<QueryResult> query(@Query(U) String username, @Query(P) String password,
-      @Query(value = Q, encoded = true) String query);
+  public Call<QueryResult> query(@Query(value = Q, encoded = true) String query);
```
Comment (fmachado) [inline src/main/java/org/influxdb/impl/InfluxDBService.java]: parameters removed: (_non-backward compatible change_)

File: src/main/java/org/influxdb/impl/InfluxDBService.java [hunk 8]
```diff
@@ -39,49 +39,44 @@
...
 
   @POST("query")
-  public Call<QueryResult> postQuery(@Query(U) String username,
-      @Query(P) String password, @Query(value = Q, encoded = true) String query);
+  public Call<QueryResult> postQuery(@Query(value = Q, encoded = true) String query);
```
Comment (fmachado) [inline src/main/java/org/influxdb/impl/InfluxDBService.java]: parameters removed: (_non-backward compatible change_)

File: src/main/java/org/influxdb/impl/InfluxDBService.java [hunk 9]
```diff
@@ -39,49 +39,44 @@
...
 
   @Streaming
   @GET("query?chunked=true")
-  public Call<ResponseBody> query(@Query(U) String username,
-      @Query(P) String password, @Query(DB) String db, @Query(value = Q, encoded = true) String query,
+  public Call<ResponseBody> query(@Query(DB) String db, @Query(value = Q, encoded = true) String query,
```
Comment (fmachado) [inline src/main/java/org/influxdb/impl/InfluxDBService.java]: parameters removed: (_non-backward compatible change_)

File: src/main/java/org/influxdb/impl/InfluxDBService.java [hunk 10]
```diff
@@ -39,49 +39,44 @@
...
       @Query(CHUNK_SIZE) int chunkSize);
 
   @Streaming
   @POST("query?chunked=true")
-  public Call<ResponseBody> query(@Query(U) String username,
-          @Query(P) String password, @Query(DB) String db, @Query(value = Q, encoded = true) String query,
+  public Call<ResponseBody> query(@Query(DB) String db, @Query(value = Q, encoded = true) String query,
```
Comment (fmachado) [inline src/main/java/org/influxdb/impl/InfluxDBService.java]: parameters removed: (_non-backward compatible change_)

Comment (majst01) [reply src/main/java/org/influxdb/impl/InfluxDBService.java]: No, InfluxDBService provides only a internal API which @lxhoan adopted, the only visible API is InfluxDB,java and that didnt change.

Or did i miss something ?

Comment (lxhoan) [reply src/main/java/org/influxdb/impl/InfluxDBService.java]: the method signatures is public however the interface (InfluxDBService) itself is visible to org.influxdb.impl only, so I don't think user can import this interface in their implementation

**Pull request merged**